File: jrand-annotation-processor/src/main/java/me/xdrop/jrand/annotation/processing/ClassIdentifier.java
package me.xdrop.jrand.annotation.processing;

import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;

import javax.lang.model.element.PackageElement;
import java.util.Objects;

public class ClassIdentifier {
    private String className;
    private PackageElement packageName;

    public ClassIdentifier(String className, PackageElement packageName) {
        this.className = className;
        this.packageName = packageName;
    }

    public String getClassName() {
        return className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

    public PackageElement getPackageName() {
        return packageName;
    }

    public void setPackageName(PackageElement packageName) {
        this.packageName = packageName;
    }

    @Override
    public int hashCode() {
        return Objects.hash(packageName.toString()) + Objects.hash(getClassName());
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof ClassIdentifier) {
            return this.packageName.toString().equals(((ClassIdentifier) obj).getPackageName().toString()) && this.className.equals(((ClassIdentifier) obj).className);
        }
        return false;
    }
}

File: jrand-annotation-processor/src/main/java/me/xdrop/jrand/annotation/processing/ClassIdentifier.kt
package me.xdrop.jrand.annotation.processing

import javax.lang.model.element.PackageElement
import javax.lang.model.element.TypeElement

data class ClassIdentifier(var typeElement: TypeElement, var packageName: PackageElement) {
    val className = typeElement.simpleName.toString()

    override fun equals(other: Any?): Boolean {
        if (other is ClassIdentifier) {
            return this.className == other.className
                    && this.packageName.toString() == other.packageName.toString()
        }
        return false
    }

    override fun hashCode(): Int {
        var result = packageName.toString().hashCode()
        result = 31 * result + className.hashCode()
        return result
    }

    override fun toString(): String {
        return this.packageName.toString().plus('.').plus(this.className)
    }
}
